python/adelphi/bin/adelphi

```diff
@@ -16,55 +39,81 @@ log = logging.getLogger('adelphi')
-# not implemented yet
-#@click.option('--output', metavar='/file/path', help='Output file. If not specified, it will write to stdout')
 @click.option('--username', help='Database username')
 @click.option('--password', help='Database password')
 @click.option('--keyspaces', help='Comma-separated list of keyspaces to include. If not specified, all keypaces will '
                          'be included, except system keypaces')
 @click.option('--rf', type=int, help='Replication factor to override original setting. Optional.')
 @click.option('--anonymize/--no-anonymize', default=True, help="Enable/disable schema anonymization")
+@click.option('--output-dir', help='Directory schema files should be written to. If not specified, it will write to stdout')
 @click.pass_context
-def export(ctx, hosts, port, username, password, keyspaces, rf, anonymize):
+def export(ctx, hosts, port, username, password, keyspaces, rf, anonymize, output_dir):
 
     ctx.ensure_object(dict)
+
+    # If output-dir is specified make sure it actually exists, is a directory and is writable
+    if output_dir is not None:
+        if not os.path.exists(output_dir):
+            log.error("Output directory " + output_dir + " does not exist")
+            exit(3)
+        if not os.path.isdir(output_dir):
+            log.error("Specified output directory " + output_dir + " exists but is not a directory")
+            exit(4)
+        if not os.access(output_dir, os.W_OK):
+            log.error("Output directory " + output_dir + " exists but is not writable")
+            exit(5)
+
     ctx.obj['hosts'] = hosts.split(',')
     ctx.obj['port'] = port
     ctx.obj['username'] = username
     ctx.obj['password'] = password
     ctx.obj['keyspace_names'] = keyspaces.split(',') if keyspaces is not None else None
     ctx.obj['rf'] = rf
     ctx.obj['anonymize'] = anonymize
+    ctx.obj['output-dir'] = output_dir
 
 
-# Function implementing the behaviours we want to execute when provided with a metadata object
+# Function implementing the behaviours we want to execute when provided with a cluster object
 # from the driver.  This function will be used to generate a lower-arity function suitable
 # for use with store.with_cluster via partial application below.
 #
 # Note that "cluster" must be the last non-keyword arg here in order to make the partial
 # application happy
-def cluster_fn(export_fn, keyspace_names, options, cluster, keyspace_validation_fn = None):
+def cluster_fn(export_fn, ctx, cluster, keyspace_validation_fn = None):
+
+    keyspace_names = ctx.obj["keyspace_names"]
+    options = {'rf': ctx.obj['rf']}
+    output_dir = ctx.obj["output-dir"]
+
     metadata = cluster.metadata
     keyspace_objects = build_keyspace_objects(keyspace_names, metadata)
+
+    if ctx.obj['anonymize']:
+        # anonymize keyspace and its children
+        for keyspace_object in keyspace_objects:
+            anonymize_keyspace(keyspace_object)
+
```

Comment:
The introduction of output to files meant that we had to do anonymization earlier in the process.  Before it was being done in the CQL/Gemini export op which happens much further down the line.  We want it to happen earlier (and in this script) so that all subsequent log statements/file names/etc. will use anonymized names.